huaweicloud/services/eg/resource_huaweicloud_eg_event_subscription.go

```diff
@@ -0,0 +1,287 @@
+							Required:     true,
+							ValidateFunc: validation.StringIsJSON,
+							Description:  "The filter rule of the event source",
+						},
+					},
+				},
+				Description: "The list of the event sources.",
+			},
+			"targets": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"provider_type": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The provider type of the event target.",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The name of the event target.",
+						},
+						"detail_name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The name (key) of the target configuration detail.",
+						},
+						"detail": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsJSON,
+							Description:  "The configuration detail of the event target, in JSON format.",
+						},
+						"transform": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsJSON,
+							Description:  "The transform configuration of the event target, in JSON format.",
+						},
+						"connection_id": {
+							Type:        schema.TypeString,
+							Optional:    true,
+							Description: "The connection ID of the EG event target.",
+						},
+						"dead_letter_queue": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Computed:     true,
+							ValidateFunc: validation.StringIsJSON,
+							Description:  "The specified queue to which failure events sent, in JSON format.",
+						},
+					},
+				},
+				Description: "The list of the event targets.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The description of the event subscription.",
+			},
+			// Attributes
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the event subscription.",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The creation time of the event subscription.",
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The latest update time of the event subscription.",
+			},
+		},
+	}
+}
+
+func unmarshalEventSubscriptionParamsters(paramName, paramVal string) map[string]interface{} {
+	parseResult := make(map[string]interface{})
+	err := json.Unmarshal([]byte(paramVal), &parseResult)
+	if err != nil {
+		log.Printf("[ERROR] Invalid type of the %s, not json format", paramName)
+	}
+	return parseResult
+}
+
+func buildEventSourcesOpts(sources []interface{}) []interface{} {
+	result := make([]interface{}, len(sources))
+	for i, val := range sources {
+		source := val.(map[string]interface{})
```

Comment:
It is better to check error when doing type conversion.
```suggestion
		if source, isMap := val.(map[string]interface{}); isMap {}
```

Reply:
done.